src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -184,61 +221,42 @@ private function adaptResourceLocators(ResourceSegmentBehavior $document, $userI
         $webspaceKey = $this->documentInspector->getWebspace($document);
         $languageCode = $this->documentInspector->getLocale($document);
 
-        foreach ($document->getChildren() as $childDocument) {
-            $childNode = $this->documentInspector->getNode($childDocument);
-
-            // determine structure
-            $templatePropertyName = $this->nodeHelper->getTranslatedPropertyName('template', $languageCode);
+        $node = $this->documentInspector->getNode($document);
+        $node->getSession()->save();
 
-            if (!$childNode->hasProperty($templatePropertyName)) {
+        foreach ($document->getChildren() as $childDocument) {
+            // skip documents without assigned resource segment
+            if (!$childDocument instanceof ResourceSegmentBehavior
+                || !$currentResourceLocator = $childDocument->getResourceSegment()) {
+                $this->adaptResourceLocators($childDocument, $userId);
                 continue;
             }
 
+            // build new resource segment based on parent changes
+            $parentUuid = $this->documentInspector->getUuid($document);
+            $childPart = $this->getChildPart($currentResourceLocator);
+            $newResourceLocator = $this->generate($childPart, $parentUuid, $webspaceKey, $languageCode);
```

Comment:
I think the URL should not be generated? This means that the last part of the URL is not reused, right?